src/wp-admin/includes/class-rollback-auto-update.php

```diff
@@ -0,0 +1,405 @@
+<?php
+/**
+ * WordPress Plugin Administration API: WP_Rollback_Auto_Update class
+ *
+ * @package WordPress
+ * @subpackage Administration
+ * @since 6.3.0
+ */
+
+/**
+ * Core class for rolling back auto-update plugin failures.
+ */
+class WP_Rollback_Auto_Update {
+
+	/**
+	 * Stores handler parameters.
+	 *
+	 * @var array
+	 */
+	private $handler_args = array();
+
+	/**
+	 * Stores successfully updated plugins.
+	 *
+	 * @var array
+	 */
+	private static $processed = array();
+
+	/**
+	 * Stores fataling plugins.
+	 *
+	 * @var array
+	 */
+	private static $fatals = array();
+
+	/**
+	 * Stores `update_plugins` transient.
+	 *
+	 * @var stdClass
+	 */
+	private static $current;
+
+	/**
+	 * Stores boolean for no error from check_plugin_for_errors().
+	 *
+	 * @var bool
+	 */
+	private $no_error = false;
```

Comment:
```suggestion
	private $update_is_safe = false;
```
It might be worth being a bit more explicit here to avoid confusion when reading implementation and thinking "no" means the number of errors, etc.

> Stores whether the update is safe to proceed without Rollback.